### PR TITLE
TW-1666: Filename does not resize and time overlapping.

### PR DESCRIPTION
**Pull request by hieutbui**
## Ticket
- #1666 

### DOD:
- [x] the filename text should be resized also
- [x] the time is not overlap with text

## Root cause
* We are using `RichText` for `filename` but not providing a `textScaler` value for it. Therefore, by default it will always be `1.0` and `noScaling`.
* Inconsistencies in padding for timeline and widget for timeline:
   * We pass `MessageTime` to message widget as `endOfBubbleWidget` to create the necessary buffering for the timeline. This `MessageTime` is wrapped inside a `TextSpan` &rarr; It will resize as the system size changes.
   * We use `MessageTime` without `TextSpan` for rendering timeline &rarr; It will not resize as the system size changes.

## Solution
- Use `Text.rich` widget instead cause by default, it uses `MediaQueryData.textScaleFactor`.
- Wrap widget for timeline inside a `TextSpan`.
- Remove fake `MessageTime` inside message body.
- Re-calculate message bubble width.

## Not handled
* Change system's font size to large and very large
* Still big gap:
    * Message contains preview link &rarr; Cause width of preview and this message will has special HTML tag (`<blockquote>`)
    * Message has reply &rarr; Cause width of reply content and this message will has special HTML tag (`<blockquote>`)

## Relevant
- https://github.com/linagora/flutter_matrix_html/pull/17

## Resolved
- Web:

https://github.com/linagora/twake-on-matrix/assets/80142234/952385ad-7d5c-455c-a4e5-1406c30c104c

- Android:

https://github.com/linagora/twake-on-matrix/assets/80142234/02240cdf-9ab7-497a-ba4a-77c1ef5bd6a8

- HTML special tag that cannot calculate width:

https://github.com/linagora/twake-on-matrix/assets/80142234/502e5be4-33e3-40f2-a9a9-a709ce59c8ee

### CODE REVIEW

**Comment by sherlockvn** [comment]
### small size text
image
### Normal text
Screenshot 2024-04-03 at 09 38 18

**Comment by hoangdat** [comment]
- [ ] time MUST BE fixed size, but message content should not overlap in time

**Comment by hoangdat** [comment]
- [ ] verify why in some case, time and message have a very big gap 
image

**Comment by sherlockvn** [comment]
image

**Comment by sherlockvn** [comment]
Screenshot 2024-04-08 at 10 58 29

**Comment by hieutbui** [comment]
> image

fixed!

**Comment by hieutbui** [comment]
Pending:
- [ ] Calculate for tag name inside message

File: lib/pages/chat/events/message/message_content_builder.dart [hunk 1]
```diff
@@ -197,23 +247,150 @@ class MessageContentBuilder extends StatelessWidget {
...
+          totalMessageWidth = lastLineWithTimeWidth;
+        }
+      }
+    }
+
+    final Map<double, bool> result = {
+      totalMessageWidth: isNeedAddNewLine,
+    };
+    return result;
+  }
```
Comment (hieutbui) [inline lib/pages/chat/events/message/message_content_builder.dart:395]: Will separate later

**Review by hoangdat — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**